grpc/diatheke.proto

```diff
@@ -28,44 +28,47 @@ service Diatheke {
   rpc Models(Empty) returns (ModelsResponse) {}
 
   // Requests a new session with the given config and returns the session
-  // ID, which is required for other rpc methods.
+  // ID, which is required for other rpc methods. After the session is
+  // created, StartSession() must be called to begin executing the Diatheke
+  // model.
   rpc NewSession(NewSessionRequest) returns (SessionID) {}
 
-  // Terminates an existing session and closes any open event streams.
+  // Begin execution of the model for the given session ID. The session's
+  // event stream should be set up prior to calling this function so that
+  // the client application can respond to any initialization events that
+  // are defined in the session's model.
+  rpc StartSession(SessionID) returns (Empty) {}
+
+  // Terminates an existing session and closes any open session streams.
   // It is an error if the SessionEndRequest has an invalid SessionID.
   rpc EndSession(SessionID) returns (Empty) {}
 
-  // Requests a new event stream for the given session.
+  // Requests a new event stream for the given session. Only one stream
+  // per session is allowed.
   rpc SessionEventStream(SessionID) returns (stream DiathekeEvent) {}
 
   // Notify Diatheke when a command has completed so that it may update
   // the dialog state. The initial command request will come as part of
-  // a DiathekeEvent. While not strictly required (depeding on the model
-  // and command), it is best practice to always call this method when a
-  // command is complete. Cases where it is required include when the 
-  // command has output parameters, or when the command is followed by 
-  // another action in the Diatheke model.
+  // a DiathekeEvent. Client applications should always call this after
+  // receiving a command event, or else the session will hang.
   rpc CommandFinished(CommandStatus) returns (Empty) {}
 
   // Begin an audio input stream for a session. The first message to
   // the server should specify the sessionID, with binary audio data pushed
   // for every subsequent message. As the audio is recognized, Diatheke
   // will respond with appropriate events on the session's event stream.
   // <p>
-  // While it is allowed to call this multiple times during a single session,
-  // clients should never have multiple audio input streams running concurrently
-  // for the same session (the audio may mix and result in unpredictable
-  // behavior). Previous audio streams should always be closed before starting
-  // a new one.
+  // Only one stream per session is allowed. Previous audio streams should
+  // always be closed before starting a new one.
```

Comment:
If only one stream per session is allowed, why would the client start a new one? How do we handle push-to-talk if it's all in one stream?

Reply:
I'll clarify. These means one stream concurrently.  You can create and destroy as many as you like, as long as there is only one at a time.

Reply:
I'll also note that even if it was a single stream allowed over the lifetime of a session, push to talk would still be theoretically possible because the client application controls when to push audio to the stream (although, at that point it is a question of whether Cubic endpointing would be able to figure out this is a new utterance).

Reply:
Yes, the problem is not whether the client can push audio when it wants to. It's a problem of pushing unconnected audio to the same recognizer.

I'll approve once you've clarified the comment.